server/src/main/java/org/elasticsearch/cluster/metadata/IndexMetadata.java

```diff
@@ -484,6 +485,9 @@ public static APIBlock readFrom(StreamInput input) throws IOException {
     @Nullable // since we store null if DataTier.TIER_PREFERENCE_SETTING failed validation
     private final List<String> tierPreference;
 
+    @Nullable
+    private final Integer shardsPerNodeLimit;
```

Comment:
I think I would prefer to use -1 as no limit since that is the definition on the setting as well. I think an explicit -1 setting value is OK, which would cause this to have a (Integer) -1 value anyway.

Perhaps we need a test with an explicit -1 to verify this case works too (seems we do not have such a test)?

Reply:
Test is not necessary thanks to your comment above :) I could just use the setting outright now because we only use the setting and no fallback to node settings so we can use `int` and have all kinds of tests that test the setting default of `-1` :)